packages/rum-core/src/domain/contexts/featureFlagContext.spec.ts

```diff
@@ -0,0 +1,154 @@
+    })
+
+    it('should not add feature flag evaluation when the ff feature_flags is disabled', () => {
+      const { lifeCycle } = setupBuilder.build()
+
+      lifeCycle.notify(LifeCycleEventType.VIEW_CREATED, {
+        startClocks: relativeToClocks(0 as RelativeTime),
+      } as ViewCreatedEvent)
+
+      featureFlagContexts.addFeatureFlagEvaluation('feature', 'foo')
+
+      const featureFlagContext = featureFlagContexts.findFeatureFlagEvaluations()!
+
+      expect(featureFlagContext).toEqual({})
+    })
+  })
+
+  describe('findFeatureFlagEvaluations', () => {
+    /**
+     * It could happen if there is an event happening just between view end and view creation
+     * (which seems unlikely) and this event would anyway be rejected by lack of view id
+     */
+    it('should return undefined when no current view', () => {
+      updateExperimentalFeatures(['feature_flags'])
+
+      setupBuilder.build()
+
+      expect(featureFlagContexts.findFeatureFlagEvaluations()).toBeUndefined()
+    })
+
+    it('should clear feature flag context on new view', () => {
+      updateExperimentalFeatures(['feature_flags'])
+
+      const { lifeCycle } = setupBuilder.build()
+
+      lifeCycle.notify(LifeCycleEventType.VIEW_CREATED, {
+        startClocks: relativeToClocks(0 as RelativeTime),
+      } as ViewCreatedEvent)
+      featureFlagContexts.addFeatureFlagEvaluation('feature', 'foo')
+      lifeCycle.notify(LifeCycleEventType.VIEW_ENDED, {
+        endClocks: relativeToClocks(10 as RelativeTime),
+      } as ViewEndedEvent)
+      lifeCycle.notify(LifeCycleEventType.VIEW_CREATED, {
+        startClocks: relativeToClocks(10 as RelativeTime),
+      } as ViewCreatedEvent)
+
+      const featureFlagContext = featureFlagContexts.findFeatureFlagEvaluations()!
+      expect(featureFlagContext).toEqual({})
+    })
+
+    it('should return the feature flag context corresponding to the start time', () => {
+      updateExperimentalFeatures(['feature_flags'])
+
+      const { lifeCycle, clock } = setupBuilder.withFakeClock().build()
+
+      lifeCycle.notify(LifeCycleEventType.VIEW_CREATED, {
+        startClocks: relativeToClocks(0 as RelativeTime),
+      } as ViewCreatedEvent)
+
+      clock.tick(10)
+      featureFlagContexts.addFeatureFlagEvaluation('feature', 'one')
+      lifeCycle.notify(LifeCycleEventType.VIEW_ENDED, {
+        endClocks: relativeToClocks(10 as RelativeTime),
+      } as ViewEndedEvent)
+      lifeCycle.notify(LifeCycleEventType.VIEW_CREATED, {
+        startClocks: relativeToClocks(10 as RelativeTime),
+      } as ViewCreatedEvent)
+
+      clock.tick(10)
+      featureFlagContexts.addFeatureFlagEvaluation('feature', 'two')
+
+      lifeCycle.notify(LifeCycleEventType.VIEW_ENDED, {
+        endClocks: relativeToClocks(30 as RelativeTime),
+      } as ViewEndedEvent)
+      lifeCycle.notify(LifeCycleEventType.VIEW_CREATED, {
+        startClocks: relativeToClocks(30 as RelativeTime),
+      } as ViewCreatedEvent)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏those statements do not seem to be useful for the test